pkg/fsm/fsm_active_file_io.go

```diff
@@ -95,6 +95,44 @@ func (f *FsMachine) saveFile(file *types.InputFile) StateFn {
 	return activeState
 }
 
+func (f *FsMachine) statFile(file *types.StatFile) StateFn {
+
+	// create the default paths
+	sourcePath := fmt.Sprintf("%s/%s/%s", file.SnapshotMountPath, "__default__", file.Filename)
+
+	l := log.WithFields(log.Fields{
+		"filename":   file.Filename,
+		"sourcePath": sourcePath,
+	})
+
+	fi, err := os.Stat(sourcePath)
```

Comment:
is this still ok if it turns out to be a folder?

Reply:
Yeah, you can stat directories - the mode field of the FileInfo tells you if it's a directory, regular file, symlink, or other weirder thing (socket, device file, etc)